### PR TITLE
Allow disabling IPv4

### PR DESCRIPTION
**Pull request by tmjd**

### CODE REVIEW

File: filesystem/etc/calico/confd/templates/bird.cfg.template [hunk 1]
```diff
@@ -3,7 +3,8 @@ include "bird_aggr.cfg";
 include "bird_ipam.cfg";
 {{$node_ip_key := printf "/host/%s/ip_addr_v4" (getenv "NODENAME")}}{{$node_ip := getv $node_ip_key}}
 
-router id {{$node_ip}};
+{{$router_id := getenv "ROUTER_ID" ""}}
```
Comment (caseydavenport) [inline filesystem/etc/calico/confd/templates/bird.cfg.template]: We should probably prefix this with `CALICO_`

File: filesystem/etc/calico/confd/templates/bird.cfg.template [hunk 2]
```diff
@@ -39,6 +40,7 @@ protocol direct {
   interface -"cali*", "*"; # Exclude cali* but include everything else.
 }
 
+{{if ne "" ($node_ip)}}# IPv4 disabled on this node.
```
Comment (caseydavenport) [inline filesystem/etc/calico/confd/templates/bird.cfg.template]: Is this comment in the right place? Won't that show up even when node_ip is defined?

File: pkg/startup/startup.go [hunk 3]
```diff
@@ -421,19 +421,24 @@ func configureIPsAndSubnets(node *api.Node) (bool, error) {
...
-	} else {
+	} else if ipv6Env != "none" {
 		if ipv6Env != "" {
 			node.Spec.BGP.IPv6Address = parseIPEnvironment("IP6", ipv6Env, 6)
 		}
 		validateIP(node.Spec.BGP.IPv6Address)
 	}
 
+	if ipv4Env == "none" && (ipv6Env == "" || ipv6Env == "none") {
+		log.Warn("No IP Addresses configured or set to autodetect")
```
Comment (caseydavenport) [inline pkg/startup/startup.go]: "No IP addresses configured, and autodetection is not enabled" is probably more clear / correct.

File: pkg/startup/startup.go [hunk 4]
```diff
@@ -421,19 +421,24 @@ func configureIPsAndSubnets(node *api.Node) (bool, error) {
...
+		terminate()
+	}
+
 	// Detect if we've seen the IP address change, and flag that we need to check for conflicting Nodes
-	if oldIpv4 == "" || node.Spec.BGP.IPv4Address != oldIpv4 {
+	if node.Spec.BGP.IPv4Address != oldIpv4 {
 		log.Info("Node IPv4 changed, will check for conflicts")
 		return true, nil
 	}
-	if (oldIpv6 == "" && node.Spec.BGP.IPv6Address != "") || (oldIpv6 != "" && node.Spec.BGP.IPv6Address != oldIpv6) {
```
Comment (caseydavenport) [inline pkg/startup/startup.go]: Weird... the old way just looks totally bonkers. Wonder how that happened.

Comment (tmjd) [reply pkg/startup/startup.go]: That's what I thought too. I look at it multiple times to figure out if it was handling some case that I wasn't thinking of.

**Pull request merged**